contentprovider.go

```diff
@@ -552,6 +552,48 @@ func findSection(secs []DocumentSection, off, sz uint32) (uint32, bool) {
 	return 0, false
 }
 
+// findMaxOverlappingSection returns the index of the section in secs that
+// overlaps the most with the area defined by off and sz, relative to the size
+// of the section. If no section overlaps, it returns 0, false. If multiple
+// sections overlap the same amount, the first one is returned.
+//
+// The implementation assumes that sections do not overlap and are sorted by
+// DocumentSection.Start.
+func findMaxOverlappingSection(secs []DocumentSection, off, sz uint32) (uint32, bool) {
+	// Find the first section that might overlap
+	j := sort.Search(len(secs), func(i int) bool { return secs[i].End > off })
+
+	if j == len(secs) || secs[j].Start >= off+sz {
+		// No overlap.
+		return 0, false
+	}
+
+	relOverlap := func(j int) float64 {
```

Comment:
I think we can be more direct than relative overlap. If I am not mistaken, we only care about partial vs full overlap? Could we not do something like:
```go
// if the 2nd section has a full overlap always return it
if j+1 < len(secs) && secs[j+1].Start >= off && secs[j+1].End <= off+sz {
  return j+1, true
}
// Otherwise the first section has equivalent scoring to other sections
return j, secs[j].Start < off+sz
```

Reply:
>  If I am not mistaken, we only care about partial vs full overlap? 

I think the amount of the overlap should count. For example, if your code looks like

```
aaaa bbbb
```

both `aaaa` and `bbbb` are symbols. If you search for `a bbb` it does make sense to prefer the second symbol over the first: the smaller the overlap is, the higher the chance that the match is low quality.

Sure, both are just partial overlaps and get the same boost for that. However, we don't just care about partial vs. complete overlap. The two symbols might be of a different kind and hence get a different score `scoreSymbolKind`.

Reply:
aah that is a good point. Almost makes me think we should be scoring every symbol match and taking the max score?